src/main/java/school/lemon/changerequest/java/io/pr1/task3/CountEngineImp.java

```diff
@@ -0,0 +1,163 @@
+            }
+        } catch (IOException n) {
+        }
+        return result;
+    }
+
+    @Override
+    public long countWords(String filename) throws FileNotFoundException {
+        File file = new File(filename);
+        if (!file.exists()) {
+            throw new FileNotFoundException();
+        }
+        char c;
+        int words = 0;
+        int charForWords = 0;
+        int spaces = 0;
+        boolean wordWasCounted = false;
+        boolean firstWord = true;
+        Pattern charsP = Pattern.compile("[a-zA-Z0-9]");
+        Pattern spacesP = Pattern.compile("[\n\\s]");
+        Matcher charsM;
+        Matcher spacesM;
+        try {
+            BufferedInputStream bufferedInputStream = new BufferedInputStream(new FileInputStream(file));
+            c = (char) bufferedInputStream.read();
+            CharSequence str = Character.toString(c);
+            while (bufferedInputStream.available() > 0) {
+                charsM = charsP.matcher(str);
+                spacesM = spacesP.matcher(str);
+                if (spacesM.matches()) {
+                        spaces++;
+                        wordWasCounted = false;
+                        charForWords = 0;
+                } else if ((charsM.matches() && (spaces > 0) || firstWord)) {
+                    firstWord = false;
+                    charForWords++;
+                    if (charForWords == 1 && !wordWasCounted) {
+                        words++;
+                        wordWasCounted = true;
+                    }
+                }
+                c = (char) bufferedInputStream.read();
+                str = Character.toString(c);
+            }
+        } catch (IOException n) {
+        }
+        return words;
+    }
+
+    @Override
+    public long countParagraphs(String filename) throws FileNotFoundException {
+        File file = new File(filename);
+        int paragraphs = 0;
+        boolean firstParagraph = true;
+        if (!file.exists()) {
+            throw new FileNotFoundException();
+        }
+        BufferedReader br = new BufferedReader(new FileReader(file));
+        try {
+            String contentLine = br.readLine();
+            while (contentLine != null) {
+                if (countWordsInLine(contentLine) > 2 && !firstParagraph) {
+                    contentLine = br.readLine();
+                    if (contentLine.isEmpty()) {
+                        contentLine = br.readLine();
+                        if (contentLine.isEmpty()) {
+                            paragraphs++;
+                        }
+                    }
+                } else if (countWordsInLine(contentLine) > 2 && firstParagraph) {
+                    paragraphs++;
+                    firstParagraph = false;
+                }
+                contentLine = br.readLine();
+            }
+
+        } catch (IOException n) {
+        }
+        return paragraphs;
+    }
+
+    private int countWordsInLine(String string) {
```

Comment:
Not sure it's really needed. Also I'd prefer to determine words count with regexp, instead of check each character.